eth_account/_utils/encode_typed_data/encoding_and_hashing.py

```diff
@@ -63,9 +63,21 @@ def encode_field(
         raise ValueError(f"Missing value for field `{name}` of type `{type_}`")
 
     elif is_array_type(type_):
-        type_ = parse_parent_array_type(type_)
-        type_value_pairs = [encode_field(types, name, type_, item) for item in value]
-        data_types, data_hashes = zip(*type_value_pairs)
+        # handle array type with non-array value
+        if not isinstance(value, list):
+            raise ValueError(
+                f"Invalid value for field `{name}` of type `{type_}`: "
+                f"expected array, got `{value}` of type `{type(value)}`"
+            )
+
+        parsed_type = parse_parent_array_type(type_)
+        type_value_pairs = [
+            encode_field(types, name, parsed_type, item) for item in value
+        ]
+        if type_value_pairs:
+            data_types, data_hashes = zip(*type_value_pairs)
+        else:
+            data_types, data_hashes = (), ()
```

Comment:
It seems like this is always a constant (`(), ()`). We should return that constant here instead and forego the need to `keccak(encode(empty_tuples))` every time this conditional is true. We can add a comment in the code to clarify what that constant means.

```py
if not type_value_pairs:
    # the keccak hash of encode(`(), ()` )
    return ("bytes32", constant_value_here)
    
data_types, data_hashes = zip(*type_value_pairs)

return return ("bytes32", keccak(encode(data_types, data_hashes)))